### PR TITLE
hir: Two preparatory changes for #120206

### PR DESCRIPTION
**Pull request by petrochenkov**
cc https://github.com/rust-lang/rust/pull/120206
r? @compiler-errors

### CODE REVIEW

**Review by compiler-errors — APPROVED** [review]

File: compiler/rustc_hir/src/intravisit.rs [hunk 1]
```diff
@@ -729,7 +726,12 @@ pub fn walk_expr<'v, V: Visitor<'v>>(visitor: &mut V, expression: &'v Expr<'v>)
         ExprKind::DropTemps(ref subexpression) => {
             visitor.visit_expr(subexpression);
         }
-        ExprKind::Let(ref let_expr) => visitor.visit_let_expr(let_expr),
+        ExprKind::Let(ref let_expr) => {
```
Comment (cjgillot) [inline compiler/rustc_hir/src/intravisit.rs:729]: Nit: could you match on struct fields too, to ensure we catch changes to the HIR tree?

**Comment by cjgillot** [comment]
CI failure.
@bors r-

**Comment by compiler-errors** [comment]
Whoops. Thanks for catching that.

**Pull request merged**